packages/hydrooj/src/handler/record.ts

```diff
@@ -308,36 +332,45 @@ class RecordDetailConnectionHandler extends ConnectionHandler {
 
     @param('rid', Types.ObjectID)
     async prepare(domainId: string, rid: ObjectID) {
+        let viewSelfCode = false;
```

Comment:
Code-only views should never connect to this endpoint to receive updates, and this should be done explicitly in frontend.